### PR TITLE
feat(amazonaurora): adding customization Aurora Vector Store

### LINKED ISSUE #599: (amazonaurora): vector store customization options
### Describe the feature

I really love the AmazonAuroraDefaultVectorStore Construct, however I have two main issues with this construct which make it really hard to use:
- it uses by default a new VPC and there is no way to customize it
- the security group created is not exposed via attributes so additional rules cannot be added

### Use Case

To have more flexibility on the Aurora Vector Store Deployment

### Proposed Solution

It would be great if we could customize the VPC via the construct props and expose some attributes like this:
```ts
/**
 * The Security Group attached to the Aurora DB Instances in the Cluster.
 */
public readonly auroraSecurityGroup: ec2.ISecurityGroup;
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by aws-rafams**
Fixes #599 
- Added VPC customization option to Aurora Vector Store
- Made some attributes accessible such as:
   - `vpc` to understand where the cluster has been deployed
   - `auroraSecurityGroup` for adding custom Ingress Rules

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Comment by krokoko** [comment]
Thanks @aws-rafams ! Did you deploy the construct to verify it works with your changes ?

**Review by krokoko — APPROVED** [review]

**Comment by aws-rafams** [comment]
Yes, I have tested it and it works, the only thing to be careful is that the VPC must have Private and Public subnets, otherwise deployment will fail. I have updated the docs accordingly.

**Review by krokoko — DISMISSED** [review]

File: src/cdk-lib/amazonaurora/aurora-default-vector-store.ts [hunk 1]
```diff
@@ -156,7 +157,7 @@ export class AmazonAuroraDefaultVectorStore extends cdk.Resource {
       clusterIdentifier: this.clusterIdentifier,
       defaultDatabaseName: this.databaseName,
       vpc: this.vpc,
-      vpcSubnets: { subnetType: ec2.SubnetType.PUBLIC },
+      vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_ISOLATED },
```
Comment (aws-rafams) [inline src/cdk-lib/amazonaurora/aurora-default-vector-store.ts:160]: I also spotted this which is not super best practice. But as you realized, changing it makes the build fail. Also if we change it we probably would need to introduce VPC Endpoints for the RDS Data API used by the Lambda function.

Comment (krokoko) [reply src/cdk-lib/amazonaurora/aurora-default-vector-store.ts]: Running another build with the change above, resolving this for now

**Comment by krokoko** [comment]
Still failing, looking at the build

**Review by scottschreckengaust — APPROVED** [review]

**Comment by krokoko** [comment]
Fixed the build, @aws-rafams could you please have a look/test those changes make sure it's all good on your end ? Then we can merge

**Comment by aws-rafams** [comment]
> Fixed the build, @aws-rafams could you please have a look/test those changes make sure it's all good on your end ? Then we can merge

will pull changes and test them. Also a question, do you think we should generate a name for the cluster identifier? Using the construct two times results in a deployment error as there will be two clusters with the same identifier (it is now internally hardcoded).

**Comment by krokoko** [comment]
> > Fixed the build, @aws-rafams could you please have a look/test those changes make sure it's all good on your end ? Then we can merge
> 
> will pull changes and test them. Also a question, do you think we should generate a name for the cluster identifier? Using the construct two times results in a deployment error as there will be two clusters with the same identifier (it is now internally hardcoded).

Sure, good catch ! We can do it in a separate PR just to avoid too many changes here if that works for you !

**Review by scottschreckengaust — APPROVED** [review]

**Review by dineshSajwan — APPROVED** [review]
LGTM

**Comment by aws-rafams** [comment]
Was testing the changes, and it seems due to a change recently introduced regarding VPC endpoints (#614), the CustomResource Lambda which gets credentials from SecretsManager and connects to the Aurora Database, is now on a `SubnetType.PRIVATE_ISOLATED` subnet rather than on a `SubnetType.PRIVATE_WITH_EGRESS`. The deployment fails as the Lambda can't reach SM. Will now test manually creating the SM endpoint, but in my opinion this should be transparent to the user and a new endpoint should be created if not present.
Screenshot 2024-08-08 at 22 24 48

**Comment by krokoko** [comment]
> Was testing the changes, and it seems due to a change recently introduced regarding VPC endpoints (#614), the CustomResource Lambda which gets credentials from SecretsManager and connects to the Aurora Database, is now on a `SubnetType.PRIVATE_ISOLATED` subnet rather than on a `SubnetType.PRIVATE_WITH_EGRESS`. The deployment fails as the Lambda can't reach SM as in the provided custom VPC there is no existing endpoint. Will now test manually creating the SM endpoint, but in my opinion this should be transparent to the user and a new endpoint should be created if not present. Screenshot 2024-08-08 at 22 24 48

Thanks @aws-rafams , I was able to reproduce. I found out a similar issue when testing another sample which uses the custom resource. I reverted the changes introduced to the VPC. We can merge your original changes which were working, and I'll have a look to re-apply later the changes from the previous PR

**Comment by krokoko** [comment]
@aws-rafams Could you please pull the latest on this branch and do a last run to make sure it works on your end ? If it is all good, we can merge. Thank you !

**Comment by aws-rafams** [comment]
> > Was testing the changes, and it seems due to a change recently introduced regarding VPC endpoints (#614), the CustomResource Lambda which gets credentials from SecretsManager and connects to the Aurora Database, is now on a `SubnetType.PRIVATE_ISOLATED` subnet rather than on a `SubnetType.PRIVATE_WITH_EGRESS`. The deployment fails as the Lambda can't reach SM as in the provided custom VPC there is no existing endpoint. Will now test manually creating the SM endpoint, but in my opinion this should be transparent to the user and a new endpoint should be created if not present. Screenshot 2024-08-08 at 22 24 48
> 
> Thanks @aws-rafams , I was able to reproduce. I found out a similar issue when testing another sample which uses the custom resource. I reverted the changes introduced to the VPC. We can merge your original changes which were working, and I'll have a look to re-apply later the changes from the previous PR


I have tested the latest commits and on my end everything works as expected.
Screenshot 2024-08-08 at 23 39 27

**Review by scottschreckengaust — APPROVED** [review]

**Review by krokoko — APPROVED** [review]

**Pull request merged**